src/main/java/com/bergerkiller/bukkit/tc/signactions/SignActionType.java

```diff
@@ -1,40 +1,43 @@
 package com.bergerkiller.bukkit.tc.signactions;
 
 public enum SignActionType {
-    NONE, REDSTONE_CHANGE, REDSTONE_ON, REDSTONE_OFF, MEMBER_ENTER, MEMBER_MOVE, MEMBER_LEAVE, GROUP_ENTER, GROUP_LEAVE,
-    MEMBER_UPDATE, GROUP_UPDATE;
+    NONE(false, true), 
+    REDSTONE_CHANGE(true, false), 
+    REDSTONE_ON(true, false), 
+    REDSTONE_OFF(true, false),  
+    MEMBER_ENTER(false, true),  
+    MEMBER_MOVE(false, true),  
+    MEMBER_LEAVE(false, true),  
+    GROUP_ENTER(false, true),  
+    GROUP_LEAVE(false, true), 
+    MEMBER_UPDATE(false, false),  
+    GROUP_UPDATE(false, false);
+    
+    
+    private signActionType(boolean redstone, boolean movement) {
+        this.redstone = redstone;
+        this.movement = movement;
+    }
 
     /**
      * This sign action type is redstone-related
      * 
      * @return redstone related
      */
-    public boolean isRedstone() {
-        switch (this) {
-        case REDSTONE_CHANGE:
-        case REDSTONE_ON:
-        case REDSTONE_OFF:
-            return true;
-        default:
-            return false;
-        }
-    }
+    private final boolean redstone;
 
     /**
      * This sign action type was generated by train or cart movement
      * 
      * @return movement related
      */
+    private final boolean movement;
```

Comment:
Javadoc broke here, and in my style I generally put all fields at the top of the class declaration not in the middle of the class body

Reply:
I already fixed the javadoc mess. I forgot about moving it to the new generated method. I will adjust the class structure as you requested it. Thank you for reviewing!